src/main/java/net/sf/jabref/LoadedDatabase.java

```diff
@@ -0,0 +1,73 @@
+package net.sf.jabref;
+
+import net.sf.jabref.model.database.BibDatabase;
+import net.sf.jabref.model.database.BibDatabaseType;
+import net.sf.jabref.model.database.BibDatabaseTypeDetection;
+
+import java.io.File;
+import java.util.Objects;
+import java.util.Vector;
+
+public class LoadedDatabase {
+
+    private static final String DATABASE_TYPE = "DATABASE_TYPE";
+
+    private final BibDatabase database;
+    private final MetaData metaData;
+
+    public LoadedDatabase() {
+        this(new BibDatabase());
+    }
+
+    public LoadedDatabase(BibDatabase database) {
+        this(database, new MetaData());
+    }
+
+    public LoadedDatabase(BibDatabase database, MetaData metaData) {
+        this.database = Objects.requireNonNull(database);
+        this.metaData = Objects.requireNonNull(metaData);
+
+        this.setType(getType());
+    }
+
+    public LoadedDatabase(BibDatabase database, MetaData metaData, File file) {
+        this(database, metaData);
+
+        this.metaData.setFile(file);
+    }
+
+    public BibDatabaseType getType() {
```

Comment:
Mhhh another type :) I would prefer a rename to getMode. If not then isBibLatexMode should be renamed to isBiblatexType.